### PR TITLE
fcitx5-rose-pine: init at 0-unstable-2024-03-01

### PR DESCRIPTION
**Pull request by RoseHobgoblin**
## Description of changes

Adds three rose-pine themes for fcitx5
https://github.com/rose-pine/fcitx5

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by itslychee** [comment]
related pr for future purposes: https://github.com/NixOS/nixpkgs/pull/292478
the author was unable to reopen it because they rewrote the history before reopening it

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -16582,6 +16582,11 @@
       fingerprint = "1401 1B63 393D 16C1 AA9C  C521 8526 B757 4A53 6236";
     }];
   };
+  RoseHobgoblin = {
```
Comment (itslychee) [inline maintainers/maintainer-list.nix:16585]: make sure to lowercase this

File: pkgs/by-name/fc/fcitx5-rose-pine/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    cp -rv rose-pine* $out/share/fcitx5/themes/
+
+    runHook postInstall
+  '';
+
+
+  meta = {
+    description = "Fcitx5 themes based on Rosé Pine";
+    homepage = "https://github.com/rose-pine/fcitx5";
+    maintainers = with lib.maintainers; [ RoseHobgoblin ];
```
Comment (itslychee) [inline pkgs/by-name/fc/fcitx5-rose-pine/package.nix:30]: and this as well so it won't break on evaluation

**Review by itslychee — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (eclairevoyant) [reply maintainers/maintainer-list.nix:16585]: > make sure to lowercase this

is there a guideline that requires this?

File: pkgs/by-name/fc/fcitx5-rose-pine/package.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+stdenvNoCC.mkDerivation {
+  pname = "fcitx5-rose-pine";
+  version = "0-unstable-2024-03-01";
+  
+  src = fetchFromGitHub {
+    owner = "rose-pine";
+    repo = "fcitx5";
+    rev = "148de09929c2e2f948376bb23bc25d72006403bc";
+    hash = "0gywb8ykb2j3zyr20w1mc3f5xfw2aw4v9b821d55w36ja753k52a";
```
Comment (eclairevoyant) [inline pkgs/by-name/fc/fcitx5-rose-pine/package.nix:14]: Please set this to an empty string, and get the hash from the error when building (currently it's not an SRI hash)

Comment (RoseHobgoblin) [reply pkgs/by-name/fc/fcitx5-rose-pine/package.nix:14]: I don't even know what that means

File: pkgs/by-name/fc/fcitx5-rose-pine/package.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    mkdir -pv $out/share/fcitx5/themes/
+    cp -rv rose-pine* $out/share/fcitx5/themes/
+
+    runHook postInstall
+  '';
+
+
+  meta = {
+    description = "Fcitx5 themes based on Rosé Pine";
+    homepage = "https://github.com/rose-pine/fcitx5";
```
Comment (eclairevoyant) [inline pkgs/by-name/fc/fcitx5-rose-pine/package.nix:29]: `meta.license` should be set, but I don't find a proper license under the upstream repo. In such cases it should be set to `lib.licenses.unfree`; see https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

**Comment by itslychee** [comment]
> is there a guideline that requires this?
@eclairevoyant github isn't allowing me to reply to this but it doesn't seem like it, mostly assumed from the widespread lowercase naming that it has and according to the `maintainers.nix` file it's preferred to keep `github == handle`

https://github.com/NixOS/nixpkgs/blob/244ee5631a7a39b0c6bd989cdf9a1326cd3c5819/maintainers/maintainer-list.nix#L34C1-L34C124

**Comment by eclairevoyant** [comment]
> > is there a guideline that requires this?
> 
> mostly assumed from the widespread lowercase naming that it has

I'd rather avoid asking for IMO arbitrary changes; there's certainly a lot of exceptions to this, to the point where I wouldn't call it even an unspoken rule.

> according to the `maintainers.nix` file it's preferred to keep `github == handle`

Yes, and their github handle is `RoseHobgoblin`; though gh is technically case-insensitive, that's how it's displayed.

**Comment by RoseHobgoblin** [comment]
Whatever. I lowercased it in the file.

File: pkgs/by-name/fc/fcitx5-rose-pine/package.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    maintainers = with lib.maintainers; [ rosehobgoblin ];
+    platforms = lib.platforms.all;
+    meta.license = lib.licenses.unfree;
```
Comment (eclairevoyant) [inline pkgs/by-name/fc/fcitx5-rose-pine/package.nix:32]:
```suggestion
    license = lib.licenses.unfree;
```

**Review by eclairevoyant — DISMISSED** [review]
One more fix. Also it seems that CI (OfBorg) is complaining about some editor config/formatting issues; please check and address that

**Comment by itslychee** [comment]
> Whatever. I lowercased it in the file.

It was a misunderstanding on my end, you did not have to change it.

**Comment by RoseHobgoblin** [comment]
> > Whatever. I lowercased it in the file.
> 
> It was a misunderstanding on my end, you did not have to change it.

There's no point in changing it back.

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -16582,6 +16582,11 @@
       fingerprint = "1401 1B63 393D 16C1 AA9C  C521 8526 B757 4A53 6236";
     }];
   };
+  rosehobgoblin = {
+    name = "J. L. Bowden";
+    github = "rosehobgoblin";
+    githubId = "84164410";
```
Comment (itslychee) [inline maintainers/maintainer-list.nix:16588]: this needs to be an integer 
https://gist.github.com/GrahamcOfBorg/14356eb528c57a9d7fa7588d2b51e087

**Review by itslychee — CHANGES_REQUESTED** [review]

**Review by itslychee — APPROVED** [review]

**Review by Gerg-L — APPROVED** [review]
Looks good, thanks for contributing to nixpkgs!

**Review by eclairevoyant — APPROVED** [review]

**Pull request merged**